### PR TITLE
Use a fallback method for the website projects full path calculation

### LINKED ISSUE #12337: WebSite projects opened from IIS fail to install packages
### NuGet Product Used

Visual Studio Package Management UI, Visual Studio Package Manager Console

### Product Version

VS 17.4

### Worked before?

17.3

### Impact

I'm unable to use this version

### Repro Steps & Context

1. In IIS Manager, create a virtual directory as an application.
2. In VS, use "open website", select "local IIS", and then select the directory
3. Try to install a package that has `lib/<tfm>/*.dll` files.

An obscure error message is reported: 

> Failed to add reference to ‘Newtonsoft.Json’.
> 
> Invalid URI: The Authority/Host could not be parsed.

### Verbose Logs

_No response_

### PR DESCRIPTION
**Pull request by nkolev92**
## Bug

Fixes: https://github.com/NuGet/Home/issues/12337

Regression? Last working version:

## Description

This is a consequence of https://github.com/NuGet/NuGet.Client/pull/4698. We're now using IVsHierarchy to init projects, but unfortunately the implementation for the full project path for website doesn't work the same when a website project is loaded from IIS as it then returns the url instead of the local path. 

A way to solve this is to special case the website check. 

## PR Checklist

- [x] PR has a meaningful title
- [x] PR has a linked issue.
- [x] Described changes

- **Tests**
  - [ ] Automated tests added
  - **OR**
  
  - [x] Test exception - Will ask for a manual test.
  - **OR**
  - [ ] N/A 

- **Documentation**
  
  - [ ] Documentation PR or issue filled
  - **OR**
  - [x] N/A

### CODE REVIEW

**Comment by zivkan** [comment]
Wow, this PR is WAY easier than what I was trying to do 🤦 

👍

**Comment by nkolev92** [comment]
> Wow, this PR is WAY easier than what I was trying to do 🤦
> 
> 👍

I'm talking to some other people to see if there's a solution that works everywhere, but I don't hate this.

File: src/NuGet.Clients/NuGet.VisualStudio.Common/IDE/VsHierarchyUtility.cs [hunk 1]
```diff
@@ -24,12 +24,21 @@ public static class VsHierarchyUtility
...
             var format = (IPersistFileFormat)project;
             format.GetCurFile(out string projectPath, out uint _);
             return projectPath;
         }
 
+        public static string GetProjectPathForWebsiteProject(IVsHierarchy project)
+        {
+            Assumes.Present(project);
+            ThreadHelper.ThrowIfNotOnUIThread();
+            project.GetCanonicalName((uint)VSConstants.VSITEMID.Root, out string projectPath);
```
Comment (jeffkl) [inline src/NuGet.Clients/NuGet.VisualStudio.Common/IDE/VsHierarchyUtility.cs:42]: Team Review: Please put some comments in here so people know why we are special casing web site projects

Comment (nkolev92) [reply src/NuGet.Clients/NuGet.VisualStudio.Common/IDE/VsHierarchyUtility.cs:42]: Added.

**Review by erdembayar — APPROVED** [review]

**Review by jeffkl — APPROVED** [review]

**Review by zivkan — APPROVED** [review]

**Pull request merged**